src/main.rs

```diff
@@ -31,11 +32,14 @@ fn main() {
     let args: Args = Docopt::new(USAGE)
         .and_then(|d| d.decode())
         .unwrap_or_else(|e| e.exit());
-    let program = match args {
+    let program = (match args {
         Args { arg_program: Some(p), .. } => Program::parse(&p),
-        Args { arg_file: Some(p), .. } => Program::from_file(p).unwrap(),
+        Args { arg_file: Some(p), .. } => Program::from_file(p),
         _ => panic!("Bad args."),
-    };
+    }).unwrap_or_else(|e| {
+        println!("FATAL: {}", e);
+        process::exit(1);
+    });
```

Comment:
@nixpulvis comments would be great on this. initially i used an intermediary `program_result` var which was unwrap_or_else'd but I thought this was better. not sure if wrapping a `match` in parens is weird though